src/java.base/share/classes/java/lang/invoke/MethodHandleNatives.java

```diff
@@ -87,7 +90,8 @@ static CallSiteContext make(CallSite cs) {
             // become unreachable, its Context is retained by the Cleanable instance
             // (which is referenced from Cleaner instance which is referenced from
             // CleanerFactory class) until cleanup is performed.
-            CleanerFactory.cleaner().register(cs, newContext);
+            new CleanerImpl.PhantomCleanableRef(cs, CleanerFactory.cleaner(),
+                    newContext, JDKResource.Priority.CALL_SITES);
```

Comment:
This is based on impl details. Is it possible to use e.g. a `CleanerFactory.criticalCleaner().register(...)` that will use different priorities?

Reply:
1) `CleanerFactory` could get a new method since it is in `jdk.internal.ref`, but such alternative implementation would spin up another thread.
2) `Cleaner` is a public interface, so I cannot add an alternative method. Regrettably it's also a final class, so I can't make CleanerFactory wrap the common cleaner and override just the `register()` method.

I find that the info that we need to do something different already leaks to MethodHandleNatives and since your suggested solution has impact on memory usage, I would rather choose breaking the encapsulation.